src/components/processes-configuration.js

```diff
@@ -98,35 +98,30 @@ const ProcessTsos = ({
         })
     );
 
-    useEffect(() => {
+    const updateProcessTsos = (processTsos) => {
         handleProcessTsosChanged(processIndex, processTsos);
-        // Do not add handleProcessTsosChanged as dependency to avoid infinite loop
-        // To be changed
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [processTsos]);
+        setProcessTsos(processTsos);
+    };
 
     const handleTsoSourcingActorChanged = (index, event) => {
-        const processTsosCopy = [...processTsos];
-        processTsosCopy[index] = {
-            id: processTsosCopy[index].id,
+        processTsos[index] = {
+            id: processTsos[index].id,
             sourcingActor: event.target.value,
         };
```

Comment:
Maybe just one line, like this : 

processTsos[index].sourcingActor: event.target.value